tests/vast-test.vala

```diff
@@ -60,14 +58,13 @@ int main (string[] args) {
         assert (0 == a.dimension);
         assert (typeof (void) == a.scalar_type);
         assert (sizeof (void) == a.scalar_size);
-        assert (null == a.shape);
-        assert (null == a.strides);
         assert (1 == a.size);
         assert (0 == a.origin);
-        assert (null == a.data);
-        assert ("dtype: void, dsize: %lu, dimension: 0, shape: (), strides: (), mem: 0B".printf (sizeof (void)) == a.to_string ());
+        assert (null != a.data);
+        //assert ("dtype: void, dsize: %lu, dimension: 0, shape: (), strides: (), mem: 0B".printf (sizeof (void)) == a.to_string ());
 
-        var b = a.reshape ({2, 2, 2, 4});
+        size_t [] shape = {2, 2, 2, 4};
+        var b = Object.new (typeof (Vast.Array), "dimension", shape.length, "shape", shape) as Vast.Array;
```

Comment:
@rainwoodman By the way, here, you can use `dimension: shape.length, shape: shape` notation. It's supported by Vala for alternating varidic arguments.